localstack-core/localstack/utils/lambda_debug_mode/lambda_debug_mode_session.py

```diff
@@ -0,0 +1,68 @@
+import logging
+from typing import Optional
+
+from localstack.aws.api.lambda_ import Arn
+from localstack.config import LAMBDA_DEBUG_MODE, LAMBDA_DEBUG_MODE_CONFIG_PATH
+from localstack.utils.lambda_debug_mode.lambda_debug_mode_config import (
+    LambdaDebugConfig,
+    LambdaDebugModeConfig,
+    load_lambda_debug_mode_config,
+)
+
+LOG = logging.getLogger(__name__)
+
+
+class _LambdaDebugModeSession:
+    _instance = None
```

Comment:
@dfangl Does this match our common way of creating singletons?

Reply:
No, we usually create singletons with a `.get()` function:

```
    @staticmethod
    @singleton_factory
    def get() -> "_LambdaDebugModeSession":
        """Returns a singleton instance of the debug mode session."""
        return _LambdaDebugModeSession()
```

I would prefer to avoid overriding `__new__` here. Instead ofusing the module global, we can always call `_LambdaDebugModeSession.get()`. Might be nice to rename it to avoid the leading `_` then, though.